File: Ackermann-function/Go/ackermann-function-1.go
package main

func main(){
    for m := 0; m < 4; m++ {
        for n := 0; n < 13; n++ {
            println(Ackermann(uint(m),uint(n)))
        }
    }
}

func Ackermann(m, n uint) uint {
	switch uint(0) {
	case m:
		return n + 1
	case n:
		return Ackermann(m - 1, 1)
	}
	return Ackermann(m - 1, Ackermann(m, n - 1))
}

File: Ackermann-function/C/ackermann-function-1.c
#include <stdio.h>

int ackermann(int m, int n)
{
        if (!m) return n + 1;
        if (!n) return ackermann(m - 1, 1);
        return ackermann(m - 1, ackermann(m, n - 1));
}

int main()
{
        int m, n;
        for (m = 0; m < 4; m++)
                for (n = 0; n < 14; n++)
                        printf("%d\n", ackermann(m, n));

        return 0;
}
